### PR TITLE
Migrate deps to go-module and adds go1.11 support

### PR DESCRIPTION
**Pull request by akashshinde**

### CODE REVIEW

File: tools.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// +build tools
+
+package tools
```
Comment (jarifibrahim) [inline tools.go]: Why do we need this package?

Comment (akashshinde) [reply tools.go]: It is required to track tool dependencies(https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module)

**Review by jarifibrahim — DISMISSED** [review]
I ran the PR locally and I was able to build the binary and run it.

**Review by jarifibrahim — CHANGES_REQUESTED** [review]
`make test` doesn't seem to work. 

```
swagger/swagger.yaml
go: finding github.com/fabric8-services/build-tool-detector/client latest
go: finding github.com/fabric8-services/build-tool-detector/tool/cli latest
go: finding github.com/fabric8-services/build-tool-detector/tool latest
go test -vet off github.com/fabric8-services/build-tool-detector github.com/fabric8-services/build-tool-detector/config github.com/fabric8-services/build-tool-detector/controllers github.com/fabric8-services/build-tool-detector/controllers/error github.com/fabric8-services/build-tool-detector/domain/repository github.com/fabric8-services/build-tool-detector/domain/repository/github github.com/fabric8-services/build-tool-detector/domain/token github.com/fabric8-services/build-tool-detector/domain/types github.com/fabric8-services/build-tool-detector/log -v
?   	github.com/fabric8-services/build-tool-detector	[no test files]
# gopkg.in/h2non/gock.v1
/home/ijarif/go/pkg/mod/gopkg.in/h2non/gock.v1@v1.0.13/matchers.go:123:13: undefined: parth.Sequent
FAIL	github.com/fabric8-services/build-tool-detector/config [build failed]
FAIL	github.com/fabric8-services/build-tool-detector/controllers [build failed]
=== RUN   TestError
Running Suite: Error Suite
==========================
```

**Comment by akashshinde** [comment]
> `make test` doesn't seem to work.
> 
> ```
> swagger/swagger.yaml
> go: finding github.com/fabric8-services/build-tool-detector/client latest
> go: finding github.com/fabric8-services/build-tool-detector/tool/cli latest
> go: finding github.com/fabric8-services/build-tool-detector/tool latest
> go test -vet off github.com/fabric8-services/build-tool-detector github.com/fabric8-services/build-tool-detector/config github.com/fabric8-services/build-tool-detector/controllers github.com/fabric8-services/build-tool-detector/controllers/error github.com/fabric8-services/build-tool-detector/domain/repository github.com/fabric8-services/build-tool-detector/domain/repository/github github.com/fabric8-services/build-tool-detector/domain/token github.com/fabric8-services/build-tool-detector/domain/types github.com/fabric8-services/build-tool-detector/log -v
> ?   	github.com/fabric8-services/build-tool-detector	[no test files]
> # gopkg.in/h2non/gock.v1
> /home/ijarif/go/pkg/mod/gopkg.in/h2non/gock.v1@v1.0.13/matchers.go:123:13: undefined: parth.Sequent
> FAIL	github.com/fabric8-services/build-tool-detector/config [build failed]
> FAIL	github.com/fabric8-services/build-tool-detector/controllers [build failed]
> === RUN   TestError
> Running Suite: Error Suite
> ==========================
> ```

I was able to reproduce this issue, It had been introduced when we use `gock v1.0.13`. works when we use `gock v.1.0.12` .

**Review by jarifibrahim — APPROVED** [review]

**Comment by kwk** [comment]
@akashshinde is it possible, that this PR caused the project to not be buildable with Go 1.10?

**Pull request merged**